types/node/url.d.ts

```diff
@@ -1,29 +1,36 @@
 declare module "url" {
     import { ParsedUrlQuery, ParsedUrlQueryInput } from 'querystring';
 
-    interface UrlObjectCommon {
-        auth?: string;
-        hash?: string;
-        host?: string;
-        hostname?: string;
-        href?: string;
-        path?: string;
-        pathname?: string;
-        protocol?: string;
-        search?: string;
-        slashes?: boolean;
-    }
-
     // Input to `url.format`
-    interface UrlObject extends UrlObjectCommon {
-        port?: string | number;
+    interface UrlObject {
+        auth?: string | null;
+        hash?: string | null;
+        host?: string | null;
+        hostname?: string | null;
+        href?: string | null;
+        path?: string | null;
+        pathname?: string | null;
+        protocol?: string | null;
+        search?: string | null;
+        slashes?: boolean | null;
+        port?: string | number | null;
         query?: string | null | ParsedUrlQueryInput;
     }
 
     // Output of `url.parse`
-    interface Url extends UrlObjectCommon {
-        port?: string;
-        query?: string | null | ParsedUrlQuery;
+    interface Url {
+        auth: string | null;
+        hash: string | null;
+        host: string | null;
+        hostname: string | null;
+        href: string | null;
```

Comment:
Could you provide case when `href` can be `null`? In the following tests `href` is always `string`:

```sh
$ node --version
v10.16.0
$ node -p 'url.parse("")'
Url {
  protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: null,
  search: null,
  query: null,
  pathname: null,
  path: null,
  href: '' }
$ nvm use 8
Now using node v8.16.2 (npm v6.4.1)
$ node -p 'url.parse("")'
Url {
  protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: null,
  search: null,
  query: null,
  pathname: null,
  path: null,
  href: '' }
```

Reply:
Good catch, I've reverted that particular change.

Reply:
Could you squash your commits into one?

Reply:
Is that really necessary? I've never had to do that before when contributing to DefinitelyTyped.

Reply:
Squash merge is the default for DT maintainers so I don't think it's necessary for contributors to squash them ahead of time.